src/v/coproc/tests/fixtures/coproc_test_fixture.cc

```diff
@@ -82,103 +102,65 @@ ss::future<> coproc_test_fixture::restart() {
-                   [&pm, m_ntp] {
-                       auto partition = pm.get(m_ntp.source_ntp());
-                       return partition && partition->is_leader()
-                              && (!m_ntp.is_materialized() || pm.log(m_ntp.input_ntp()));
-                   })
-            .then([&pm, m_ntp, limit, offset, timeout] {
-                auto partition = pm.get(m_ntp.source_ntp());
-                auto partition_proxy = make_partition_proxy(
-                  m_ntp, partition, pm);
-                return do_drain(
-                         offset,
-                         limit,
-                         timeout,
-                         [pp = std::move(partition_proxy)](
-                           model::offset next_offset) mutable {
-                             return pp->make_reader(log_rdr_cfg(next_offset));
-                         })
-                  .then([](auto data) {
-                      return std::optional<model::record_batch_reader::data_t>(
-                        std::move(data));
-                  });
-            });
-      });
+    co_return events;
 }
 
 ss::future<model::offset> coproc_test_fixture::push(
   const model::ntp& ntp, model::record_batch_reader rbr) {
-    auto shard_id = _root_fixture->app.shard_table.local().shard_for(ntp);
-    if (!shard_id) {
-        vlog(
-          coproc::coproclog.error,
-          "No ntp exists, cannot push data to log: {}",
-          ntp);
-        return ss::make_ready_future<model::offset>(model::offset(-1));
+    vlog(coproc::coproclog.info, "About to produce to ntp: {}", ntp);
+    auto results = co_await std::move(rbr).for_each_ref(
+      kafka_publish_consumer(
+        *_client, model::topic_partition{ntp.tp.topic, ntp.tp.partition}),
+      model::no_timeout);
+    for (const auto& r : results) {
+        vassert(
+          r.error_code != kafka::error_code::unknown_topic_or_partition,
+          "Coproc test harness expects input logs to already be in "
+          "existance, "
+          "use setup() before starting your test");
     }
-    vlog(
-      coproc::coproclog.info,
-      "Pushing record_batch_reader to ntp: {} on shard_id: {}",
-      ntp,
-      *shard_id);
-    return _root_fixture->app.partition_manager.invoke_on(
-      *shard_id,
-      [ntp, rbr = std::move(rbr)](cluster::partition_manager& pm) mutable {
-          auto partition = pm.get(ntp);
-          return tests::cooperative_spin_wait_with_timeout(
-                   60s,
-                   [partition] { return partition && partition->is_leader(); })
-            .then([rbr = std::move(rbr), partition]() mutable {
-                return partition
-                  ->replicate(
-                    std::move(rbr),
-                    raft::replicate_options(
-                      raft::consistency_level::quorum_ack))
-                  .then([](auto r) { return r.value().last_offset; });
-            });
-      });
+    co_return results.back().base_offset;
```

Comment:
Did you want `base_offset` or `last_offset`?

Reply:
I just left this as a placeholder, as the only test that checked the return value of `push` was temporarily removed, and this functions return value will be changed in a later commit to return `results` anyway.